src/main/java/com/aws/greengrass/shadowmanager/sync/CloudDataClient.java

```diff
@@ -242,8 +242,16 @@ private void handleUpdate(MqttMessage message) {
         String thingName = shadowRequest.getThingName();
         String shadowName = shadowRequest.getShadowName();
         logger.atDebug().kv(LOG_THING_NAME_KEY, thingName).kv(LOG_SHADOW_NAME_KEY, shadowName)
-                .log("Received cloud update request");
-        syncHandler.pushLocalUpdateSyncRequest(thingName, shadowName, message.getPayload());
+                .log("Received cloud update sync request");
+        try {
+            syncHandler.pushLocalUpdateSyncRequest(thingName, shadowName, message.getPayload());
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            logger.atError()
+                    .setEventType(LogEvents.CLOUD_DATA_CLIENT_SUBSCRIPTION_ERROR.code())
+                    .setCause(e)
```

Comment:
I don't think it makes sense to set the cause here - printing an InterruptedException is not generally useful. You could include in the error log that it was interrupted while adding the local sync request